### DISCUSSION TITLE
Astro can also be headless CMS

### DISCUSSION THREAD

**Discussion by xiyuvi** [withastro/roadmap#637] [Proposal]
### Body

I want to create a static JSON data from the article list and Markdown document, which can be used by other external programs。
I hope astro can support this feature, so that astro can play the role of headless CMS. Thank you very much

**Comment by jasikpark** [comment]
@tony-sull has a bit of experience with this idea I believe, with his two repo blog setup.

**Comment by tony-sull** [comment]
Yep, this should be doable today with an API route :eyes:

```ts
// pages/api/posts.json.ts

import type { APIRoute } from "astro";
import { getCollection } from "astro:content";

export const get: APIRoute = async () => {
    const allPosts = await getCollection("blog");

    const data = allPosts.map((post) => {
        // return whatever JSON data you need from each post's frontmatter
        return {
            title: post.data.title,
            summary: post.data.summary,
            published: post.data.pubDate,
        }
    })
    
    // if the project is using an SSR adapter, you could also return a `Response` object
    return {
        body: JSON.stringify(data)
    }
}
```

**Reply by xiyuvi** [reply]
Thank you for your answer. Astro is really powerful, but unfortunately, I don't understand the code. I look forward to future authors with blog themes enabling this feature
